File: python/study/basic/return_example.py
# -*- coding: utf-8 -*-
"""
文 件 名: return_example.py
文件描述: 在 Python 中，return 语句只能在函数内部使用，而不能在全局作用域中使用
作    者: HanKin
创建日期: 2024.10.12
修改日期：2024.10.12

Copyright (c) 2024 HanKin. All rights reserved.
"""
import sys

def download_files():
    # 下载文件的逻辑
    print("Downloading files...")

def upload_files():
    # 上传文件的逻辑
    print("Uploading files...")
    print(x)

if __name__ == '__main__':
    argv_cnt = len(sys.argv)
    if argv_cnt == 2:
        print('download files')
        download_files()
        #return     # SyntaxError: 'return' outside function
        sys.exit()  # 使用 sys.exit() 退出程序
    # __main__函数是全局作用域
    x = 3158
    print('upload files')
    upload_files()

'''
(base) D:\Github\Storage\python\study\basic>python return_example.py
upload files
Uploading files...
3158
'''

File: c++/project/webserver/cgi/return_example.c
/*******************************************************************************
* 文 件 名: return_example.c
* 文件描述: 居然可以加括号，这两种形式是等价的，编译器会将它们视为相同的语义
           这种灵活性可能源于 C 语言的设计历史，早期的 C 语言受到 B 语言等前辈语言的影响，允许这种书写风格
* 备    注: return 语句后的表达式可以加括号，也可以不加括号，这是语法允许的两种写法，两者在功能上没有任何区别
* 作    者: HanKin
* 创建日期: 2025.09.18
* 修改日期：2025.09.18
*
* Copyright (c) 2025 HanKin. All rights reserved.
*******************************************************************************/
#include <stdio.h>

int test(int x)
{
    if (x == 1) return 0;
    return(1);
}

int main()
{
    int ret = test(1);
    printf("ret = %d\n", ret);
    ret = test(0);
    printf("ret = %d\n", ret);
    return 0;
}
/*
root@hankin:~/opencv# ./a.out 
ret = 0
ret = 1
*/